packages/daylight/start.go

```diff
@@ -257,69 +205,81 @@ func Start() {
 	}()
 
 	Exit := func(code int) {
-		model.GormClose()
 		delPidFile()
-		os.Exit(code)
+		model.GormClose()
 		statsd.Close()
+		os.Exit(code)
 	}
 
-	readConfig()
-
-	if len(config.ConfigIni["db_type"]) > 0 {
-		// The installation process is already finished (where user has specified DB and where wallet has been restarted)
-		err = model.GormInit(
-			config.ConfigIni["db_host"], config.ConfigIni["db_port"],
-			config.ConfigIni["db_user"], config.ConfigIni["db_password"], config.ConfigIni["db_name"])
+	initGorm := func(dbCfg conf.DBConfig) {
+		err = model.GormInit(dbCfg.Host, dbCfg.Port, dbCfg.User, dbCfg.Password, dbCfg.Name)
 		if err != nil {
-			log.WithFields(log.Fields{"db_user": config.ConfigIni["db_user"], "db_password": config.ConfigIni["db_password"],
-				"db_name": config.ConfigIni["db_name"], "type": consts.DBError}).Error("can't init gorm")
+			log.WithFields(log.Fields{
+				"db_user": dbCfg.User, "db_password": dbCfg.Password, "db_name": dbCfg.Name, "type": consts.DBError,
+			}).Error("can't init gorm")
 			Exit(1)
 		}
 	}
 
-	// create first block
-	if *utils.GenerateFirstBlock == 1 {
-		log.Info("Generating first block")
-		parser.FirstBlock()
-		fmt.Println(*utils.FirstBlockNodePublicKey)
-		os.Exit(0)
+	conf.ParseFlags()
+
+	if conf.NoConfig() {
+		conf.WebInstall = true
+	} else {
+		// override default data
+		if err := conf.LoadConfig(); err != nil {
+			log.Error("loadConfig:", err)
+			return
+		}
 	}
 
-	log.WithFields(log.Fields{"work_dir": *utils.Dir, "version": consts.VERSION}).Info("started with")
+	conf.OverrideFlags()
 
-	// kill previously run apla
-	if !utils.Mobile() {
-		killOld()
-	}
+	if *conf.FlagReinstall {
+		if err := conf.SaveConfig(); err != nil {
+			log.Error("saveConfig:", err)
+			Exit(1)
+		}
 
-	// TODO: ??
-	if fi, err := os.Stat(*utils.Dir + `/logo.png`); err == nil && fi.Size() > 0 {
-		utils.LogoExt = `png`
+		if err := model.InitDB(conf.Config.DB); err != nil {
+			log.Error("initDB:", err)
```

Comment:
what's a shitty log is here? it should be formatted like others, log.WithFields(log.Fields{"type": DBError}) and so on